src/main/java/org/jenkinsci/plugins/workflow/steps/BodyExecution.java

```diff
@@ -49,31 +51,41 @@ public abstract class BodyExecution implements Future<Object>, Serializable {
     public abstract Collection<StepExecution> getCurrentExecutions();
 
     /**
-     * Attempts to cancel an executing body block.
-     *
-     * <p>
-     * If the body has finished executing, or is cancelled already, the attempt will
-     * fail. This method is asynchronous. There's no guarantee that the cancellation
-     * has happened or completed before this method returns.
-     *
-     * @return false if the task cannot be cancelled.
+     * @deprecated use {@link #cancel(Throwable)} with {@link FlowInterruptedException}
      */
-    public abstract boolean cancel(CauseOfInterruption... causes);
+    @Deprecated
+    public boolean cancel(CauseOfInterruption... causes) {
+        if (Util.isOverridden(BodyExecution.class, getClass(), "cancel", Throwable.class)) {
+            return cancel(new FlowInterruptedException(Result.ABORTED, true, causes));
+        } else {
+            throw new AbstractMethodError("Override cancel(Throwable) from " + getClass());
+        }
+    }
 
     /**
-     * @deprecated
-     *      Use other overloaded forms of the cancel method to provide richer context.
+     * @deprecated use {@link #cancel(Throwable)} to provide richer context
      */
+    @Deprecated
     public boolean cancel(boolean b) {
-        return cancel(new Exception());
+        return cancel(new FlowInterruptedException(Result.ABORTED, true));
     }
 
     /**
-     * Convenience method around {@link #cancel(CauseOfInterruption...)} in case
-     * the cause is a random exception.
+     * Attempts to cancel an executing body block.
+     *
+     * <p>
+     * If the body has finished executing, or is cancelled already, the attempt will
+     * fail. This method is asynchronous. There's no guarantee that the cancellation
+     * has happened or completed before this method returns.
+     *
+     * @return false if the task cannot be cancelled.
      */
     public boolean cancel(Throwable t) {
```

Comment:
Maybe worth adding a note to the Javadoc that callers should probably pass a `FlowInterruptedException` here.

Reply:
Could also inside the method check instance of throwable against FlowInterruptedException and if false wrap it?

Reply:
I would rather than not wrap a non-`FlowInterruptedException`. There is no real reason to.

Reply:
(See also https://github.com/jenkinsci/workflow-cps-plugin/pull/570#discussion_r927712779.)